geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/stats/DescriptiveStats.scala

```diff
@@ -16,22 +16,22 @@ import scala.collection.JavaConverters._
 import scala.collection.immutable.ListMap
 import scala.Array._
 
-class DescriptiveStats(val attributes: Seq[Int]) extends Stat with Serializable {
+case class DescriptiveStats(val attributes: Seq[Int]) extends Stat with Serializable {
```

Comment:
Yeah, I think we should not switch everything to case classes. Just create a new stat in the `newcopy` method. Basically copy the `+` method, but just add this stat to the new one. Using case classes is a) exposing some variables that we don't want to be public and b) sharing some mutable state (which I assume is not the goal)